core_backend/app/contents/routers.py

```diff
@@ -158,6 +165,209 @@ async def retrieve_content_by_id(
+                    description="Please upload a CSV file.",
+                )
+            ]
+        )
+        raise HTTPException(status_code=400, detail=error_list_model.dict())
+
+    df = _load_csv(file)
+    await _csv_checks(df=df, user_id=user_db.user_id, asession=asession)
+
+    # Add each row to the content database
+    content_list = []
+    for _, row in df.iterrows():
+        content = ContentCreate(
+            content_title=row["content_title"],
+            content_text=row["content_text"],
+            content_language=row.get("content_language", "ENGLISH"),
+            content_tags=[],
+            content_metadata=row.get("content_metadata", {}),
+        )
+        content_db = await save_content_to_db(
+            user_id=user_db.user_id, content=content, asession=asession
+        )
+        content_retrieve = _convert_record_to_schema(content_db)
+        content_list.append(content_retrieve)
+
+    return content_list
+
+
+def _load_csv(file: UploadFile) -> pd.DataFrame:
+    """
+    Load the CSV file into a pandas DataFrame
+    """
+
+    try:
+        df = pd.read_csv(file.file, dtype=str)
+    except EmptyDataError as e:
+        error_list_model = CustomErrorList(
+            errors=[
+                CustomError(
+                    type="empty_data",
+                    description="The CSV file is empty",
+                )
+            ]
+        )
+        raise HTTPException(status_code=400, detail=error_list_model.dict()) from e
+    except ParserError as e:
+        error_list_model = CustomErrorList(
+            errors=[
+                CustomError(
+                    type="parse_error",
+                    description="CSV is unreadable (parsing error)",
+                )
+            ]
+        )
+        raise HTTPException(status_code=400, detail=error_list_model.dict()) from e
+    except UnicodeDecodeError as e:
+        error_list_model = CustomErrorList(
+            errors=[
+                CustomError(
+                    type="encoding_error",
+                    description="CSV is unreadable (encoding error)",
+                )
+            ]
+        )
+        raise HTTPException(status_code=400, detail=error_list_model.dict()) from e
+    if df.empty:
+        error_list_model = CustomErrorList(
+            errors=[
+                CustomError(
+                    type="no_rows_csv",
+                    description="The CSV file is empty",
+                )
+            ]
+        )
+        raise HTTPException(status_code=400, detail=error_list_model.dict())
+
+    return df
+
+
+async def _csv_checks(df: pd.DataFrame, user_id: int, asession: AsyncSession) -> None:
+    """
+    Perform checks on the CSV file to ensure it meets the requirements
+    """
+
+    # check if content_title and content_text columns are present
+    cols = df.columns
+    error_list = []
+    if "content_title" not in cols or "content_text" not in cols:
```

Comment:
Maybe strip all texts in the db and csv just to be sure.

Reply:
ahh for duplicate check? will do